spring/boot2-webflux-autoconfigure/src/main/java/com/linecorp/armeria/spring/web/reactive/ArmeriaReactiveWebServerFactory.java

```diff
@@ -221,6 +255,69 @@ public WebServer getWebServer(HttpHandler httpHandler) {
         return armeriaWebServer;
     }
 
+    private static List<ServerPort> armeriaPorts(ServerBuilder sb) {
+        try {
+            final Field ports = ServerBuilder.class.getDeclaredField("ports");
+            ports.setAccessible(true);
+            @SuppressWarnings("unchecked")
+            final List<ServerPort> armeriaPorts = (List<ServerPort>) ports.get(sb);
+            return armeriaPorts;
+        } catch (NoSuchFieldException | IllegalAccessException ignored) {
+            throw new Error(); // Should never reach here.
+        }
+    }
+
+    private static boolean isArmeriaSslEnabled(@Nullable ArmeriaSettings armeriaSettings) {
+        if (armeriaSettings != null) {
+            final com.linecorp.armeria.spring.Ssl ssl = armeriaSettings.getSsl();
+            if (ssl != null) {
+                return ssl.isEnabled();
+            }
+        }
+        return false;
+    }
+
+    private static boolean isArmeriaCompressionEnabled(@Nullable ArmeriaSettings armeriaSettings) {
+        if (armeriaSettings != null) {
+            final ArmeriaSettings.Compression compression = armeriaSettings.getCompression();
+            if (compression != null) {
+                return compression.isEnabled();
+            }
+        }
+        return false;
+    }
+
+    private void configureTls(ServerBuilder sb, com.linecorp.armeria.spring.Ssl ssl) {
+        final SslStoreProvider provider = getSslStoreProvider();
+        final Supplier<KeyStore> keyStoreSupplier;
+        final Supplier<KeyStore> trustStoreSupplier;
+        if (provider != null) {
+            keyStoreSupplier = () -> {
+                try {
+                    return provider.getKeyStore();
+                } catch (Exception e) {
+                    throw new IllegalStateException(e);
+                }
+            };
+            trustStoreSupplier = () -> {
+                try {
+                    return provider.getTrustStore();
+                } catch (Exception e) {
+                    throw new IllegalStateException(e);
```

Comment:
Maybe we can just use `Exceptions.throwUnsafely()` because this will be invoked via Spring framework anyway?